exporter/diagnostic_data_collector_test.go

```diff
@@ -88,13 +88,24 @@ func TestDiagnosticDataCollectorWithCompatibleMode(t *testing.T) {
 		return
 	}
 
+	vendor, err := getVendor(t, client, ctx, logger)
+	if err != nil {
+		assert.Fail(t, err.Error())
+		return
+	}
+
+	vendorStr := ""
+	if vendor != "" {
+		vendorStr = fmt.Sprintf(",vendor=\"%s\"", vendor)
+	}
+
 	c := newDiagnosticDataCollector(ctx, client, logger, true, ti)
 
 	// The last \n at the end of this string is important
 	expected := strings.NewReader(fmt.Sprintf(`
 	# HELP mongodb_version_info The server version
 	# TYPE mongodb_version_info gauge
-	mongodb_version_info{edition="Community",mongodb="%s"} 1`, serverVersion) + "\n")
+	mongodb_version_info{edition="Community",mongodb="%s"%s} 1`, serverVersion, vendorStr) + "\n")
```

Comment:
Let's be consistent about how label and values are generated.

Reply:
Updated